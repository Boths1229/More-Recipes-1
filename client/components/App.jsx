/* eslint-disable react/prefer-stateless-function */
import React, { PropTypes } from 'react';


const propTypes = {
  children: PropTypes.shape().isRequired
};

/**
 * @description A class to mount all component
 *
 * @extends Component
 */
class App extends React.Component {
  /**
   * @memberof App
   *
   * @returns {void} returns the component to be mounted
  */
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = propTypes;

export default App;
