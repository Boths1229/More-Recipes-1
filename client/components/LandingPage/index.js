import React from 'react';
import Header from './header';
import Carousel from './carousel';
import AboutSection from './aboutSection';
import Footer from '../common/footer';

/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class LandingPage extends React.Component {
  render() {
    return (
      <div>
        <Header/>
        <Carousel/>
        <AboutSection/>
        <Footer/>
      </div>
      
    );
  }
}

export default LandingPage;