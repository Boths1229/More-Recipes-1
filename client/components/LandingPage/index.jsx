import React from 'react';
import Header from './Header';
import Carousel from './Carousel';
import AboutSection from './AboutSection';
import Footer from '../common/Footer';


/* eslint-disable react/prefer-stateless-function */

/**
 * @description A class to mount all components related to LandingPage
 *
 * @extends Component
 */
class LandingPage extends React.Component {
  /**
   * @description renders the components
   *
   * @memberof LandingPage
   *
   * @returns {JSX} returns the components
  */
  render() {
    return (
      <div>
        <Header />
        <Carousel />
        <AboutSection />
        <Footer />
      </div>
    );
  }
}

export default LandingPage;
