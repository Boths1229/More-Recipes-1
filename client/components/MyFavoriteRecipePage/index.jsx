import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Spinner } from 'react-preloading-component';
import Header from '../common/AuthHeader';
import Footer from '../common/Footer';
import FavRecipeCard from './FavoriteRecipeCard';
import { getUserFavRecipes } from '../../actionsCreator/recipes';
import verifyUser from '../../util/Authentication';


const propTypes = {
  actions: PropTypes.shape({
    getUserFavRecipes: PropTypes.func
  }).isRequired,
  user: PropTypes.shape({
    id: PropTypes.number,
    username: PropTypes.string
  }).isRequired,
  favoriteRecipes: PropTypes.oneOfType([
    PropTypes.shape(),
    PropTypes.arrayOf(PropTypes.shape())
  ]).isRequired,
  isLoading: PropTypes.bool
};

const defaultProps = {
  isLoading: null
};

/**
 * @description A class to mount all components related to MyFavoriteRecipePage
 * @extends Component
 */
export class MyFavoriteRecipePage extends Component {
  /**
   * handles rendering favorite recipe
   * @param {object} props
  */
  constructor(props) {
    super(props);

    this.renderFavoriteRecipes = this.renderFavoriteRecipes.bind(this);
  }

  /**
   * @description performs an action right after the component mount
   *
   * @memberof MyFavoriteRecipePage
   *
   * @returns {void}
  */
  componentDidMount() {
    if (verifyUser() === true) {
      this.props.actions.getUserFavRecipes(this.props.user.id);
    }
  }

  /**
   * @memberof MyFavoriteRecipePage
   *
   * @returns {void}
  */
  renderFavoriteRecipes() {
    const {
      isLoading,
      favoriteRecipes,
      user
    } = this.props;
    if (isLoading) {
      return (
        <div style={{
          marginTop: '150px',
          marginLeft: 'auto',
          marginRight: 'auto',
        }}
        >
          <Spinner />
        </div>
      );
    }
    if (favoriteRecipes.message || favoriteRecipes.favorited.length === 0) {
      return (
        <h1 style={{ margin: '150px auto 0', width: '100%', textAlign: 'center' }}>
          You have no favorite recipes
        </h1>
      );
    }
    return favoriteRecipes.favorited.map(favorites => (
      <FavRecipeCard
        key={favorites.id}
        recipe={favorites.Recipe}
        favorite={this.favoriteRecipe}
        userId={user.id}
      />
    ));
  }

  /**
   * @description renders the components
   *
   * @memberof MyFavoriteRecipePage
   *
   * @returns {JSX} returns the components
  */
  render() {
    return (
      <div>
        <Header user={this.props.user} />
        <div className="container">
          <div className="row section-popular">
            <div className="col-sm-12">
              <h2 className="pt-3 headline centered">My Favorite Recipes</h2>
            </div>
          </div>
          <div className="row pt-4">
            {
              this.renderFavoriteRecipes()
            }
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

/**
 * @description maps state to properties of MyFavoriteRecipePage
 *
 * @param  {object} state
 *
 * @returns {object} returns the state to be mapped to props
 */
function mapStateToProps(state) {
  return {
    user: state.auth.user,
    favoriteRecipes: state.recipes.favoriteRecipes,
    isLoading: state.recipes.isLoading
  };
}

/**
 * @description maps action to properties of MyFavoriteRecipePage
 *
 * @param  {object} dispatch
 *
 * @returns {object} returns the action to be bind
 */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getUserFavRecipes
    }, dispatch)
  };
}

MyFavoriteRecipePage.propTypes = propTypes;
MyFavoriteRecipePage.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(MyFavoriteRecipePage);
