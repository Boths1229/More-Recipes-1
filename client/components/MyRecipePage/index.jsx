/* eslint-disable no-unused-vars */
/**
 *  @fileOverview renders the user recipes page
 *
 *  @author       Sanni Taiwo
 *
 *  @requires     NPM:react
 *  @requires     NPM:react-redux
 *  @requires     NPM:redux
 *  @requires     NPM:toastr
 */
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import Header from '../common/authHeader';
import Footer from '../common/footer';
import MyRecipeCard from './MyRecipeCard';
import EditRecipeModal from './EditRecipeModal';
import DeleteRecipeModal from './deleteRecipeModal';
import { logoutAction } from '../../actions/authAction';
import verifyUser from '../../util/Authentication';
import {
  getUserRecipes,
  editRecipeAction,
  deleteRecipeAction,
} from '../../actions/recipesAction';

const propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number,
    username: PropTypes.string
  }),
  actions: PropTypes.shape({
    logoutAction: PropTypes.func,
    getUserRecipes: PropTypes.func,
    deleteRecipeAction: PropTypes.func,
    editRecipeAction: PropTypes.func
  }).isRequired,
  userRecipes: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};
const defaultProps = {
  user: null
};


/**
 * @description A class to mount all components related to MyRecipePage
 * @extends Component
 */
class MyRecipePage extends Component {
  /**
   * handles editing recipe
   * @param {object} props
   * @param {object} context
  */
  constructor(props, context) {
    super(props, context);
    this.state = {
      recipe: {
        recipeName: '',
      },
      index: '',
      errorMessage: ''
    };

    this.renderUserRecipes = this.renderUserRecipes.bind(this);
    this.setRecipe = this.setRecipe.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onFocus = this.onFocus.bind(this);
    this.handleDeletion = this.handleDeletion.bind(this);
  }

  /**
   * @description performs an action right after the component mount
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  componentDidMount() {
    if (verifyUser() === true) {
      this.props.actions.getUserRecipes(this.props.user.id);
    }
  }

  /**
   * @description removes errorMessage when focused
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  onFocus(event) {
    this.setState({ errorMessage: '' });
  }

  /**
   * @description get the current recipe from the user recipes
   *
   * @param {number} index
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  setRecipe(index) {
    this.setState({
      recipe: this.props.userRecipes[index],
      index,
    });
  }

  /**
   * @description handles the deletion of recipe
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleDeletion(event) {
    const { recipe, index } = this.state;
    this.props.actions.deleteRecipeAction(recipe.id, index)
      .then(() => {
        toastr.success('Recipe successfully deleted');
        $('#deleteModal').modal('hide');
      });
  }

  /**
   * @description handles the submission of form
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleSubmit(event) {
    const { recipe, index } = this.state;
    event.preventDefault();
    this.props.actions.editRecipeAction(recipe.id, recipe, index)
      .then(() => {
        toastr.success('Recipe successfully updated');
        $('#editModal').modal('hide');
      })
      .catch((err) => {
        const { data } = err.response;
        /* eslint-disable array-callback-return */
        data.map((message) => {
          this.setState({
            errorMessage: message
          });
        });
      });
  }


  /**
   * @description checks for update in form entry
   *
   * @param {object} event
   *
   * @memberof MyRecipePage
   *
   * @returns {void}
  */
  handleChange(event) {
    const { name, value } = event.target;
    this.setState({
      recipe: {
        ...this.state.recipe,
        [name]: value
      }
    });
  }


  /**
   * @description checks if a user have recipes
   *
   * @memberof MyRecipePage
   *
   * @returns {void} returns a components
  */
  renderUserRecipes() {
    if (this.props.userRecipes.length === 0) {
      return (
        <h1>
          You have no Recipe
        </h1>
      );
    }
    return this.props.userRecipes.map((recipes, i) => (
      <MyRecipeCard
        key={recipes.id}
        index={i}
        recipeDetails={recipes}
        getRecipe={this.setRecipe}
      />
    ));
  }

  /**
   * @description renders the components
   *
   * @memberof MyRecipePage
   *
   * @returns {void} returns the components
  */
  render() {
    return (
      <div>
        <Header user={this.props.user} />
        <div className="container">
          <div className="row section-popular">
            <div className="col-sm-12">
              <h2 className="pt-3 headline centered">My Recipes</h2>
            </div>
          </div>
          <div className="row pt-4">
            { this.renderUserRecipes()}
          </div>
          <EditRecipeModal
            details={this.state.recipe}
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmit}
            errorMessage={this.state.errorMessage}
            onFocus={this.onFocus}
          />
          <DeleteRecipeModal
            handleDeletion={this.handleDeletion}
          />
        </div>
        <Footer />
      </div>
    );
  }
}

MyRecipePage.propTypes = propTypes;

MyRecipePage.defaultProps = defaultProps;

MyRecipePage.contextTypes = {
  router: PropTypes.object
};

/**
 * @description maps state to properties of MyRecipePage
 *
 * @param  {object} state
 *
 * @returns {object} returns the state to be mapped to props
 */
function mapStateToProps(state) {
  return {
    user: state.auth.user,
    userRecipes: state.recipes.userRecipes
  };
}

/**
 * @description maps action to properties of MyRecipePage
 *
 * @param  {object} dispatch
 *
 * @returns {object} returns the action to be bind
 */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      getUserRecipes,
      editRecipeAction,
      deleteRecipeAction,
      logoutAction
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MyRecipePage);
