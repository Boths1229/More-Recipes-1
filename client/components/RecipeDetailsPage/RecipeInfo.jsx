import React, { PropTypes } from 'react';
import Tooltip from 'react-tooltip';

const propTypes = {
  recipeDetails: PropTypes.shape({
    recipeName: PropTypes.string,
    description: PropTypes.string,
    ingredients: PropTypes.string,
    method: PropTypes.string,
    upvotes: PropTypes.number,
    downvotes: PropTypes.number,
    image: PropTypes.string,
  }).isRequired,
  upvote: PropTypes.func.isRequired,
  downvote: PropTypes.func.isRequired,
  favorite: PropTypes.func.isRequired,
  favorited: PropTypes.bool.isRequired,
};

/* eslint-disable react/no-array-index-key  */

const RecipeInfo = props => (
  <div>
    <img className="card-img-top" src={props.recipeDetails.image} alt="Card pic cap" style={{ maxHeight: '350px' }} />
    <div className="card-body">
      <h4 className="card-title">{props.recipeDetails.recipeName}</h4>
      <p className="card-text">
        {props.recipeDetails.description}
      </p>
      <hr />
      <div>
        <h6>Ingredients</h6>
        <ul>
          <div className="row">
            {
              props.recipeDetails.ingredients.split(',').map((ingredients, i) => (
                <li
                  className="col-md-6"
                  style={{ paddingTop: '5px', fontSize: '18px' }}
                  key={i}
                >
                  {ingredients}
                </li>
              ))
            }
          </div>

        </ul>
      </div>
      <hr />
      <div>
        <h6>
          Direction
        </h6>
        <ol>
          {
            props.recipeDetails.method.split('\n').map((method, i) => (
              <li key={i}>
                {method}
              </li>
            ))
          }
        </ol>
      </div>
      <hr />
      <div style={{ display: 'inline-flex' }}>
        <div>
          <button
            className="btn vote-button"
            type="button"
            onClick={props.upvote}
            data-tip
            data-for="upvote"
          >
            <i className="fa fa-thumbs-up fa-lg" aria-hidden="true">
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>
                {props.recipeDetails.upvotes}
              </span>
            </i>
          </button>
          <Tooltip
            id="upvote"
            type="dark"
            place="top"
          >
            <span>
              upvote recipe
            </span>
          </Tooltip>
        </div>
        <div style={{ marginLeft: '8px', marginRight: '5px' }}>
          <button
            data-tip
            data-for="downvote"
            className="btn vote-button"
            type="button"
            onClick={props.downvote}
          >
            <i className="fa fa-thumbs-down fa-lg" aria-hidden="true">
              <span style={{ marginLeft: '5px', marginRight: '5px' }}>
                {props.recipeDetails.downvotes}
              </span>
            </i>
          </button>
          <Tooltip
            id="downvote"
            type="dark"
            place="bottom"
          >
            <span>
              downvote recipe
            </span>
          </Tooltip>
        </div>
        <div style={{ marginLeft: '8px', marginRight: '5px' }}>
          <button
            className="btn vote-button"
            type="button"
            onClick={props.favorite}
            data-tip
            data-for="favorite"
          >
            <i
              className="fa fa-heart"
              style={
                props.favorited ? { color: 'white' } : { color: 'black' }
              }
            />
          </button>
          <Tooltip
            id="favorite"
            type="dark"
            place="right"
          >
            <span>
              favorite recipe
            </span>
          </Tooltip>
        </div>
      </div>
      <hr />
    </div>
  </div>
);

RecipeInfo.propTypes = propTypes;

export default RecipeInfo;
