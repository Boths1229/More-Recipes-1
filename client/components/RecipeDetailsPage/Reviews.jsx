import React, { PropTypes } from 'react';
import moment from 'moment';


const propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  reviewError: PropTypes.string,
};

const defaultProps = {
  reviewError: null,
};

const Review = props => (
  <section className="border-recipes">
    <h4 className="section-reviews centered" style={{ backgroundColor: '#2b3034' }}>Reviews</h4>
    {
      !props.reviews.length ?
        <h6 style={{ textAlign: 'center' }}>This recipe have not been reviewed</h6>
      :
        <ul style={{ maxHeight: '300px', overflowY: 'scroll' }}>
          {
        props.reviews.map(review => (
          <li key={review.id} className="review-list">
            <div>
              <div >
                <div className="header">
                  <strong className="primary-font username">{review.username}</strong>
                  <strong
                    className="primary-font pull-right"
                    style={{ paddingRight: '10px' }}
                  >
                    {moment(new Date(review.createdAt)).fromNow()}
                  </strong>
                </div>
                <p>
                  { review.review }
                </p>
                <hr />
              </div>
            </div>
          </li>
        ))
      }
        </ul>
  }
    <form id="review-form" onSubmit={props.handleSubmit}>
      <div className="form-group">
        <label htmlFor="comment">Post a review</label>
        <textarea
          name="review"
          className="form-control"
          rows="5"
          id="comment"
          onChange={props.handleChange}
          onFocus={props.onFocus}
          required
        />
        {
          props.reviewError &&
          <div style={{ textAlign: 'center' }} className="alert alert-danger">
            {props.reviewError}
          </div>
        }
      </div>
      <button type="submit" id="reviewButton" className="btn vote-button pull-right">Submit</button>
    </form>
  </section>
);
Review.propTypes = propTypes;
Review.defaultProps = defaultProps;

export default Review;
