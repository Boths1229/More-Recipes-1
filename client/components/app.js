import React from 'react';

/* eslint-disable no-unused-vars */
/* eslint-disable require-jsdoc */
class App extends React.Component {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  }
}

export default App;
