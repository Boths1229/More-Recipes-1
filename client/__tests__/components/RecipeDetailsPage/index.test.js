import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from 'react-preloading-component';
import { RecipeDetailsPage } from '../../../components/RecipeDetailsPage';
import RecipeInfo from '../../../components/RecipeDetailsPage/RecipeInfo';
import Reviews from '../../../components/RecipeDetailsPage/Reviews';
import MockData from '../../__mocks__/actions/recipes';

let props;

const setup = () => {
  props = {
    actions: {
      getRecipeData: () => {},
      resetReviewError: () => {},
      postReview: () => {},
      voteARecipe: () => {},
      favoriteRecipe: () => {},
    },
    params: {
      recipeId: 2
    },
    recipeDetails: MockData.recipeData,
    isLoading: false,
    user: MockData.userData.profile,
    error: null,
    reviewError: '',
    favoriteIds: MockData.userFavoriteIds.recipeIds
  };

  return shallow(<RecipeDetailsPage {...props} />);
};

localStorage.clear = () => {};
localStorage.setItem = () => {};
localStorage.getItem = () => {};


describe('RecipeDetailsPage Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(RecipeInfo).length).toBe(1);
    expect(wrapper.find(Reviews).length).toBe(1);
  });

  it('should submit review', () => {
    const wrapper = setup();
    wrapper.setState({ review: 'This is awesome' });
    const action = wrapper.instance();
    const submitReview = jest.spyOn(action, 'handleSubmit');
    action.handleSubmit({ preventDefault: () => {} });
    expect(submitReview).toBeCalled();
    expect(wrapper.state('review')).toBe('');
  });

  it('should clear error when input field is focused on', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const clearError = jest.spyOn(action, 'onFocus');
    action.onFocus({ preventDefault: () => {} });
    expect(clearError).toBeCalled();
  });

  it('should set review when review input changes', () => {
    const wrapper = setup();
    const event = {
      target: { name: 'review', value: 'awesome stuff' }
    };

    const action = wrapper.instance();
    const setReview = jest.spyOn(action, 'handleChange');
    action.handleChange(event);
    expect(setReview).toBeCalled();
    expect(wrapper.state('review')).toBe('awesome stuff');
  });

  it('should upvote recipe when upvote method is called', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const upvoteRecipe = jest.spyOn(action, 'upvoteARecipe');
    action.upvoteARecipe();
    expect(upvoteRecipe).toBeCalled();
  });

  it('should downvote recipe when downvote method is called', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const downvoteRecipe = jest.spyOn(action, 'downvoteARecipe');
    action.downvoteARecipe();
    expect(downvoteRecipe).toBeCalled();
  });

  it('should favorite recipe when favorite method is called', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const favoriteRecipe = jest.spyOn(action, 'favoriteRecipe');
    action.favoriteRecipe();
    expect(favoriteRecipe).toBeCalled();
  });

  it('should render loader when `isLoading` is true', () => {
    const wrapper = setup();
    wrapper.setProps({ isLoading: true });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Spinner).length).toBe(1);
  });

  it('should render error when error exists', () => {
    const wrapper = setup();
    wrapper.setProps({ error: 'Recipe not found' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toBe('404 Not found');
  });
});
