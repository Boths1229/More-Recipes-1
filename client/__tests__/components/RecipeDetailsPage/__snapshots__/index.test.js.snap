// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsPage Component should render correctly 1`] = `
<div>
  <Connect(AuthHeader)
    user={
      Object {
        "createdAt": "2018-02-19T15:54:05.322Z",
        "email": "tester@test.com",
        "id": 1,
        "image": null,
        "username": "sannytee",
      }
    }
  />
  <div
    className="container recipe-details"
  >
    <section
      className="row"
    >
      <div
        className="col-sm-2"
      />
      <div
        className="col-sm-8"
      >
        <div
          className="card w-100"
          style={
            Object {
              "width": "20rem",
            }
          }
        >
          <RecipeInfo
            downvote={[Function]}
            favorite={[Function]}
            favorited={true}
            img="test"
            recipeDetails={
              Object {
                "addedBy": "sannytee",
                "createdAt": "2018-02-20T10:11:25.020Z",
                "description": "amazing",
                "downvotes": 1,
                "id": 27,
                "image": "https://firebasestorage.googleapis.com/v0/b/more-recipes-3be20.appspot.com/o/images%2F702446f3-a2c2-48c7-a9d1-af20be9281d7.jpeg?alt=media&token=1827adb2-4108-4939-9fe0-778e27dd5212",
                "ingredients": "awesome",
                "mealType": "breakfast",
                "method": "great",
                "recipeName": "new recipe",
                "reviews": Array [
                  Object {
                    "createdAt": "2018-03-11T06:20:58.315Z",
                    "recipeId": 27,
                    "review": "hi there
",
                    "userId": 5,
                    "username": "sannikay",
                  },
                ],
                "updatedAt": "2018-03-06T11:08:34.739Z",
                "upvotes": 0,
                "userId": 1,
              }
            }
            upvote={[Function]}
          />
          <Review
            handleChange={[Function]}
            handleSubmit={[Function]}
            onFocus={[Function]}
            reviewError=""
            reviews={
              Array [
                Object {
                  "createdAt": "2018-03-11T06:20:58.315Z",
                  "recipeId": 27,
                  "review": "hi there
",
                  "userId": 5,
                  "username": "sannikay",
                },
              ]
            }
          />
        </div>
      </div>
    </section>
  </div>
  <Footer />
</div>
`;

exports[`RecipeDetailsPage Component should render error when error exists 1`] = `
<div>
  <Connect(AuthHeader)
    user={
      Object {
        "createdAt": "2018-02-19T15:54:05.322Z",
        "email": "tester@test.com",
        "id": 1,
        "image": null,
        "username": "sannytee",
      }
    }
  />
  <h1
    style={
      Object {
        "paddingTop": "150px",
      }
    }
  >
    404 Not found
  </h1>
  <Footer />
</div>
`;

exports[`RecipeDetailsPage Component should render loader when \`isLoading\` is true 1`] = `
<div>
  <Connect(AuthHeader)
    user={
      Object {
        "createdAt": "2018-02-19T15:54:05.322Z",
        "email": "tester@test.com",
        "id": 1,
        "image": null,
        "username": "sannytee",
      }
    }
  />
  <div
    style={
      Object {
        "paddingTop": "350px",
      }
    }
  >
    <n
      color="#FF4601"
      secondaryColor="#e6e6e6"
      size={40}
    />
  </div>
  <Footer />
</div>
`;
